packages/@aws-cdk/aws-ecr/lib/public-repository.ts

```diff
@@ -0,0 +1,428 @@
+    // parse the name from an ARN using CloudFormation's split/select.
+    if (Token.isUnresolved(publicRepositoryArn)) {
+      throw new Error('"publicRepositoryArn" is a late-bound value, and therefore "publicRepositoryName" is required. Use `fromPublicRepositoryAttributes` instead');
+    }
+
+    const publicRepositoryName = publicRepositoryArn.split('/').slice(1).join('/');
+
+    class Import extends PublicRepositoryBase {
+      public publicRepositoryName = publicRepositoryName;
+      public publicRepositoryArn = publicRepositoryArn;
+
+      public addToResourcePolicy(_statement: iam.PolicyStatement): iam.AddToResourcePolicyResult {
+        // dropped
+        return { statementAdded: false };
+      }
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Import a public repository that resides in the same account
+   * as the current stack from repository name.
+   */
+  public static fromPublicRepositoryName(scope: Construct, id: string, publicRepositoryName: string): IPublicRepository {
+    class Import extends PublicRepositoryBase {
+      public publicRepositoryName = publicRepositoryName;
+      public publicRepositoryArn = PublicRepository.arnForLocalRepository(publicRepositoryName, scope);
+
+      public addToResourcePolicy(_statement: iam.PolicyStatement): iam.AddToResourcePolicyResult {
+        // dropped
+        return { statementAdded: false };
+      }
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Returns an ECR ARN for a repository that resides in the same account as the current stack.
+   */
+  public static arnForLocalRepository(publicRepositoryName: string, scope: IConstruct): string {
+    return Stack.of(scope).formatArn({
+      region: '',
+      service: 'ecr-public',
+      resource: 'repository',
+      resourceName: publicRepositoryName,
+    });
+  }
+
+  private static validateRepositoryName(physicalName: string) {
+    const repositoryName = physicalName;
+    if (!repositoryName || Token.isUnresolved(repositoryName)) {
+      // the name is a late-bound value, not a defined string,
+      // so skip validation
+      return;
+    }
+
+    const errors: string[] = [];
+
+    // Rules codified from https://docs.aws.amazon.com/AmazonECRPublic/latest/APIReference/API_CreateRepository.html#ecrpublic-CreateRepository-request-repositoryName
+    if (repositoryName.length < 2 || repositoryName.length > 205) {
+      errors.push('Repository name must be at least 2 and no more than 205 characters');
+    }
+    const isPatternMatch = /^(?:[a-z0-9]+(?:[._-][a-z0-9]+)*\/)*[a-z0-9]+(?:[._-][a-z0-9]+)*$/.test(repositoryName);
+    if (!isPatternMatch) {
+      errors.push('Repository name must follow the specified pattern: (?:[a-z0-9]+(?:[._-][a-z0-9]+)*/)*[a-z0-9]+(?:[._-][a-z0-9]+)*');
+    }
+
+    if (errors.length > 0) {
+      throw new Error(`Invalid ECR repository name (value: ${repositoryName})${EOL}${errors.join(EOL)}`);
+    }
+  }
```

Comment:
This validation is really similar to the `Repository.validateRepositoryName`. I think we should use the [template pattern](https://en.wikipedia.org/wiki/Template_method_pattern) here as well.

Reply:
It is similar. Same argument about multiple inheritance.

I can extract it into separate `base-repository.ts` method if you want.